camply/search/search_going_to_camp.py

```diff
@@ -0,0 +1,217 @@
+            Saturday nights)
+        campgrounds: Union[List[int], int]
+            Campground ID or List of Campground IDs
+        nights: int
+            minimum number of consecutive nights to search per campsite,defaults to 1
+        """
+        self.provider = GoingToCampProvider
+        super(SearchGoingToCamp, self).__init__(
+            provider=GoingToCampProvider(),
+            search_window=search_window,
+            weekends_only=weekends_only,
+            nights=nights,
+        )
+
+        self._recreation_area_id = self._validate_rec_area(recreation_area)
+        self._campgrounds = campgrounds
+        self.weekends_only = weekends_only
+        assert (
+            any(
+                [
+                    campsites not in [[], None],
+                    campgrounds not in [[], None],
+                    recreation_area is not None,
+                ]
+            )
+            is True
+        )
+        self.campsites = make_list(campsites)
+        self.equipment_id = self._validate_equipment(
+            equipment_id, self._recreation_area_id
+        )
+        self.campgrounds = self._get_searchable_campgrounds()
+
+    @classmethod
+    def _validate_rec_area(cls, recreation_area: int) -> int:
+        if recreation_area in [(), [], None]:
+            logger.error("At least one --rec-area must be provided")
+            exit(1)
+
+        if not isinstance(recreation_area, List) or len(recreation_area) > 1:
+            logger.error(
+                "Going To Camp only allows a single recreation area to be searched at a time"
+            )
+            exit(1)
+
+        return int(recreation_area[0])
+
+    @classmethod
+    def _validate_equipment(cls, equipment_id: Optional[int], rec_area: int):
+        if not equipment_id:
+            return
+
+        try:
+            return int(equipment_id)
+        except ValueError:
+            logger.error(
+                "Invalid equipment ID. Use the follwoing to get list of "
+                "equipment types: "
+                f"`camply equipment-types --provider goingtocamp --rec-area {rec_area}`"
+            )
+            exit(1)
+
+    def get_all_campsites(self) -> List[AvailableCampsite]:
+        """
+        Search for all campsites matching search criteria.
+
+        Because Going To Camp has no efficient way of filtering
+        campsites for multiple campgrounds, this function is limited to listeing
+        all campsites _within_ a cammpground. Iterating through all sites for
+        all campgrounds would likely lead to a buse complaints.
+
+        Returns
+        -------
+        List[AvailableCampsite]
+        """
+        available_sites = []
+        for campground in self.campgrounds:
+            sites = self.campsite_finder.list_site_availability(
+                campground,
+                self.search_window[0].start_date,
+                self.search_window[0].end_date,
```

Comment:
This smells. It looks like we should be iterating over all search windows. I wasn't aware that multiple search windows were a valid use case.

Reply:
There are also a few of properties set on the `BaseCampingSearch` instance that might be useful: https://github.com/juftin/camply/blob/173ee4792dbee1fb922438477697780631986932/camply/search/base_search.py#L90-L92

- `self.search_dates` = An array of all Possible Dates from Among Your Search Window that are today or in the future 
- `self.search_months` = An array of all of the months that comprise the search window expressed as a first of the month dt object
- `self.search_nights` = The number of nights to search - this is also validated so if you tell camply that you want to search for weekends only but you want to stay for 6 nights, it's adjusted to 5 - or if your search window is 3 days wide but you ask for 4 nights